File: find_the_highest_altitude.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
       vector<int> nums;
        int count = 0;
        nums.push_back(count);

        for (int i = 0; i < gain.size(); i++) {
            count += gain[i];
            nums.push_back(count);
        }

        int maxAltitude = -99999999;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] >= maxAltitude) {
                maxAltitude = nums[i];
            }
        }

        return maxAltitude;
    }
};

int main(){
  int x[5] = {-5,1,5,0,-7};
    vector<int> gains(x, x + sizeof x / sizeof x[0]) ;

auto result = Solution().largestAltitude(gains);
cout<<result;
}

File: Find the Highest Altitude.dart
class Solution {
  int largestAltitude(List<int> gain) {
    List<int> nums = [];
    int count = 0;
      nums.add(count);
    for (int i = 0; i < gain.length; i++) {
      count += gain[i];
      nums.add(count);
    }
    count = -99999999;
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] >= count) {
        count = nums[i];
      }
    }
    return count;
  }
}

void main() {
  var result = Solution().largestAltitude([
    44,
    32,
    -9,
    52,
    23,
    -50,
    50,
    33,
    -84,
    47,
    -14,
    84,
    36,
    -62,
    37,
    81,
    -36,
    -85,
    -39,
    67,
    -63,
    64,
    -47,
    95,
    91,
    -40,
    65,
    67,
    92,
    -28,
    97,
    100,
    81
  ]);
  print(result);
}
